packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -78,7 +78,11 @@ function getNackReconnectInfo(nackContent: INackContent) {
 const createReconnectError = (fluidErrorCode: string, err: any) =>
     wrapError(
         err,
-        (errorMessage: string) => new GenericNetworkError(fluidErrorCode, errorMessage, true /* canRetry */),
+        (errorMessage: string) => new GenericNetworkError(
+            fluidErrorCode,
+            errorMessage,
+            err?.canRetry === true || err?.canRetry === undefined, // unless explicitly specified, this will retry
```

Comment:
^^ We would always retry, regardless of the type of error. Not sure if this has always been intentional or not..

Reply:
Here is the story :)
In ODSP, we definitely want to reconnect always, because we may get disconnect with 403 due to token expiration. 403 is critical error (i.e., in general it's a game over event), but across layers we always do one retry with refreshed token to ensure host has a chance to provide new token.

With that said, I think that's the wrong layer to participate in this game. I.e. ODSP has to project such errors as recoverable. I'd need to look at the code to say if it's the case. And not sure about FRS.

Also, worth noting that for long period of time we consider errors without canRetry to be recoverable. We changed that (maybe 6 months back) - any exception, whether it's a bug in our code, or in host code (like token callback) is catastrophic. 

So, to summarize, this code is likely wrong. But I do not think you are making it any more right :)

Reply:
I don't believe there is a viable way right now for propagating non-retriable errors from the driver to the container. Maybe a TTL on the error itself? What are your thoughts?